libtransmission/web.cc

```diff
@@ -511,6 +511,10 @@ class tr_web::Impl
         (void)curl_easy_setopt(e, CURLOPT_NOSIGNAL, 1L);
         (void)curl_easy_setopt(e, CURLOPT_PRIVATE, &task);
         (void)curl_easy_setopt(e, CURLOPT_IPRESOLVE, task.ipProtocol());
+        if (mediator.proxyUrl().has_value() && !mediator.proxyUrl().value().empty())
+        {
+            (void)curl_easy_setopt(e, CURLOPT_PROXY, mediator.proxyUrl().value().data());
+        }
```

Comment:
This isn't quite right; there's no guarantee that a `std::string_view` is going to be zero-terminated but `CURLOPT_PROXY` needs a zero-terminated C string.

This one in the PR _is_ because we know it's a std::string owned by tr_session -- but still this goes against the idea of std::string_view. IMO make a private field in tr_web::Impl, `std::string proxy_url`, sibling to `cookie_file` and `user_agent` around line 386, and initialize it in the constructor same as how `user_agent` is. Then this block at 514-516 should use that string.

Reply:
After long research I found next. Settings in session loaded from file after WebMediator and tr_web created. In this case proxy_url as private field of tr_web::Impl always is empty string.

I make creation tr_web-object (property tr_session::web_) after tr_session::setSettings.